app/javascript/components/main-menu/helpers.js

```diff
@@ -21,3 +21,18 @@ export const handleUnsavedChanges = (type) => {
   }
   return window.miqCheckForChanges();
 };
+
+export const saveVerticalMenuState = (isVerticalMenuCollapsed) => {
+  window.localStorage.setItem('patternfly-navigation-primary', isVerticalMenuCollapsed ? 'collapsed' : 'expanded');
+};
+
+export const adaptContentWidth = (isVerticalMenuCollapsed) => {
+  const content = window.document.getElementsByClassName('container-pf-nav-pf-vertical-with-sub-menus')[0];
+  if (content) {
+    if (isVerticalMenuCollapsed) {
+      content.classList.add('collapsed-nav');
```

Comment:
It's a bit unfortunate we have to manipulate these classes manually, since this is a react component and the data is in state. Any way to use react here? :)

Reply:
Oh, never mind, that particular element is in `app/views/layouts/_content.html.haml` and `app/views/layouts/_center_div_*.html.haml`, and is still outside of the react navbar.

So, this makes sense for now, no issues :)

Reply:
yeah you would have to have the whole layout as a react component. No way around it for now i am afraid